File: backend/models/post.go
package models

// 文章目录
type Toc struct {
	ID    string `json:"id"`
	Depth uint   `json:"depth"`
	Title string `json:"title"`
}

// 一个章节的信息
type Chapter struct {
	MetaData  `json:"metadata"`
	Documents []MetaData `json:"documents"`
	Children  []Chapter  `json:"children"`
}

File: frontend/src/models/post.ts
import { MetaData } from './base'

export class Toc {
    id: string = ''
    depth: number = 0
    title: string = ''
}

export class Chapter {
    metadata: MetaData = new MetaData()
    documents: MetaData[] = []
    filepath: string = ''
    children: Chapter[] = []
}

export class ChapterData extends Chapter {
    collapsed: boolean = false // 前端独立属性
    id: number = 0
}
